### ISSUE TITLE
ParamValidationError when creating/updating lambda environment variable from serverless-localstack

### ISSUE THREAD

**Issue by kristiannotari** [localstack/localstack#2659]
# Type of request: This is a ...

[x] bug report
[ ] feature request

# Detailed description

I have a serverless + serverless Localstack setup, and Localstack installed locally with docker. No problems deploying to AWS. When deploying locally, during the update stack phase, an error occurs, which is the following (last piece reported here):

<details>
  <summary>Localstack error</summary>

```
File "/usr/lib/python3.8/_collections_abc.py", line 762, in __iter__
    yield self._mapping[key]
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/moto/cloudformation/parsing.py", line 744, in __getitem__
    new_output = parse_output(
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_starter.py", line 429, in parse_output
    result = parse_output_orig(output_logical_id, output_json, resources_map)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/moto/cloudformation/parsing.py", line 445, in parse_output
    output_json = clean_json(output_json, resources_map)
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_starter.py", line 199, in clean_json
    result = clean_json_orig(resource_json, resources_map)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/moto/cloudformation/parsing.py", line 286, in clean_json
    cleaned_val = clean_json(value, resources_map)
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_starter.py", line 199, in clean_json
    result = clean_json_orig(resource_json, resources_map)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/moto/cloudformation/parsing.py", line 179, in clean_json
    resource = resources_map[resource_json["Ref"]]
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/moto/cloudformation/parsing.py", line 499, in __getitem__
    new_resource = parse_and_create_resource(
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_starter.py", line 215, in parse_and_create_resource
    return _parse_and_create_resource(
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_starter.py", line 244, in _parse_and_create_resource
    resource_tuple = parsing.parse_resource(logical_id, resource_json, resources_map)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/moto/cloudformation/parsing.py", line 341, in parse_resource
    resource_json = clean_json(resource_json, resources_map)
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_starter.py", line 199, in clean_json
    result = clean_json_orig(resource_json, resources_map)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/moto/cloudformation/parsing.py", line 286, in clean_json
    cleaned_val = clean_json(value, resources_map)
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_starter.py", line 199, in clean_json
    result = clean_json_orig(resource_json, resources_map)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/moto/cloudformation/parsing.py", line 286, in clean_json
    cleaned_val = clean_json(value, resources_map)
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_starter.py", line 199, in clean_json
    result = clean_json_orig(resource_json, resources_map)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/moto/cloudformation/parsing.py", line 179, in clean_json
    resource = resources_map[resource_json["Ref"]]
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/moto/cloudformation/parsing.py", line 499, in __getitem__
    new_resource = parse_and_create_resource(
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_starter.py", line 215, in parse_and_create_resource
    return _parse_and_create_resource(
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_starter.py", line 340, in _parse_and_create_resource
    result = deploy_func(logical_id, resource_map_new, stack_name=stack_name)
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1070, in deploy_resource
    return execute_resource_action(resource_id, resources, stack_name, ACTION_CREATE)
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1100, in execute_resource_action
    result = configure_resource_via_sdk(resource_id, resources, resource_type, func, stack_name)
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1177, in configure_resource_via_sdk
    raise e
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1174, in configure_resource_via_sdk
    result = function(**params)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/client.py", line 316, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/client.py", line 607, in _make_api_call
    request_dict = self._convert_to_request_dict(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/client.py", line 655, in _convert_to_request_dict
    request_dict = self._serializer.serialize_to_request(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/validate.py", line 297, in serialize_to_request
    raise ParamValidationError(report=report.generate_report())
botocore.exceptions.ParamValidationError: Parameter validation failed:
Invalid type for parameter Environment.Variables.AWS_NODEJS_CONNECTION_REUSE_ENABLED, value: 1, type: <class 'int'>, valid types: <class 'str'>
```
</details>

The error seems to be related to the AWS_NODEJS_CONNECTION_REUSE_ENABLED environment variable serverless is trying to add to my lambdas during the updating of the stack, with value 1 (a number, which seems correct, but I am not sure 100%). Instead, Localstack internally seems to want that value to be a string. I tested the creation and updating of stack manually with aws cli using the templated generated from serverless (the same used during the automatic deploy) and changing the value from integer to a string (still "1" value) seems to work fine. 

This is the environment variable I'm talking about:
https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/node-reusing-connections.html

And this is the piece of template with the environment variable defined as number:

<details>
  <summary>Serverless update template</summary>

```json
{
...
    "SomeNameFunction": {
      "Type": "AWS::Lambda::Function",
      "Properties": {
        "Code": {
          "S3Bucket": "__local__",
          "S3Key": "some/path"
        },
        "FunctionName": "some-name",
        "Handler": "some/path",
        "MemorySize": 1024,
        "Role": {
          "Fn::GetAtt": [
            "IamRoleLambdaExecution",
            "Arn"
          ]
        },
        "Runtime": "nodejs12.x",
        "Timeout": 6,
        "Environment": {
          "Variables": {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": 1
          }
        },
        "VpcConfig": {
          "SecurityGroupIds": [
            "somesecuritygroup"
          ],
          "SubnetIds": [
            "somesubnet"
          ]
        }
      },
      "DependsOn": [
        "InputLogGroup"
      ]
    }
...
}
```
</details>

## Expected behavior

The error should not be thrown, since the value is ?correct?

## Actual behavior

The error is thrown.

# Steps to reproduce

1. Create a serverless project
2. Install, add and configure the serverless-localstack plugin
3. Define at least one lambda function on nodejs inside the serverless.yaml file
4. Start a deploy to Localstack

...or you can manually do it by creating and updating the stack via aws-cli using the templates generated from serverless without actually doing the deploy automatically.

I have used the following serverless-localstack configuration:
<details>
  <summary>serverless-localstack config</summary>

```yaml
custom:
  localstack:
    debug: true
    stages:
      # list of stages for which the plugin should be enabled
      - local
    host: http://localhost # optional - LocalStack host to connect to
    autostart: true # optional - start LocalStack in Docker on Serverless deploy
    lambda:
      # Enable this flag to improve performance
      mountCode: True
    docker:
      # Enable this flag to run "docker ..." commands as sudo
      sudo: False
```
</details>

## Command used to start LocalStack

`localstack start` or by using the automatic way with serverless-localstack deploy

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)

For manually creating and updating the stack I've used:

`aws --endpoint-url=http://localhost:4566 cloudformation create-stack --template-body file://.serverless/cloudformation-template-create-stack.json --stack-name beppe-local`

`aws --endpoint-url=http://localhost:4566 cloudformation update-stack --template-body file://.serverless/cloudformation-template-update-stack.json --stack-name beppe-local`

**Comment by leungas** [comment]
I also caught in this problem too. Not sure this helps but it seems to work ok when I shifted back to image 0.11.3.

**Comment by whummer** [comment]
Thanks for reporting @kristiannotari . This should be fixed in #2939 - can you please pull the `latest` Docker image and give it another try? Please report here if the problem persists. Thanks
